qiskit/__init__.py

```diff
@@ -14,12 +14,25 @@
 
 """Main Qiskit public functionality."""
 
+from importlib.metadata import version, PackageNotFoundError
 import pkgutil
 import sys
 import warnings
 
 import qiskit._accelerate
 
+try:
+    version("qiskit")
+except PackageNotFoundError:
+    warnings.warn(
+        "The `qiskit` package is not installed, only `qiskit-terra` is installed. "
+        "Starting in Qiskit 1.0.0 only the `qiskit` package will be published. Migrate "
+        "any requirements files still using `qiskit-terra` to use `qiskit` instead. Also "
+        "when upgrading ensure you create a new venv instead of trying to "
+        "upgrade in place.",
+        FutureWarning,
+        stacklevel=2,
+    )
```

Comment:
We can probably bikeshed around the exact message here, but my thoughts reading it were that it's not immediately clear to me what the remediation steps are (they're implied, not stated, for a user making their own venv rather than a library author).  I'm thinking perhaps we should have an external documentation link / Qiskit blog post / whatever explaining this is in more detail that we link to from the message?

Reply:
I think linking to some place with more detail makes sense. I didn't want to write 10k words about the upgrade in a warning message, so linking to a location that has full details makes a lot of sense.